packages/realm_dart/lib/src/realm_object.dart

```diff
@@ -701,7 +730,10 @@ class RealmObjectChanges<T extends RealmObjectBase> implements Finalizable {
   /// The property names that have changed.
   List<String> get properties {
     final propertyKeys = realmCore.getObjectChangesProperties(_handle);
-    return object.realm.getPropertyNames(object.runtimeType, propertyKeys);
+    return object.realm
+        .getPropertyNames(object.runtimeType, propertyKeys)
+        .map((e) => object.objectSchema.firstWhere((element) => element.mapTo == e || element.name == e).name)
+        .toList();
```

Comment:
Not sure this is the way we want to do this.

Reply:
While potentially inefficient, it is correct, and good enough for now.